### ISSUE TITLE
Beam Search extend features size

### ISSUE THREAD

**Issue by mokadyr** [facebookresearch/mmf#497]
## ❓ Questions and Help

Hi,
For Inference over butd captioning model
When I use Beam Search with batch_size=1 I get Features size (5,100,2048) while there is only 100 bounding boxes, So the attention size is (5,100) which is what I would expect for batch_size=5.
I'm not sure why it is necessary, Beam Search should affect only the decoding right?

Using greedy or nucleus sampling decoding the Features size is (1,100,2048) as expected.

**Comment by vedanuj** [comment]
Yes you are right .. it is not necessary to expand the feature dimension to 5. We do this in order to treat each path as a separate sample but it can be optimized to use just the same feature. Would you be willing to submit a PR?

We also have a batch beam search implementation in progress which will land soon and we won't be restricted to use batch size 1. [Here](https://github.com/facebookresearch/mmf/pull/477)
